code/modules/detectivework/scanner.dm

```diff
@@ -135,8 +135,14 @@
 			LAZYADD(det_data[DETSCAN_CATEGORY_FINGERS], md5(scanned_human.dna?.unique_identity))
 
 	else if(!ismob(scanned_atom))
-
-		det_data[DETSCAN_CATEGORY_FINGERS] = GET_ATOM_FINGERPRINTS(scanned_atom)
+		//Bubberstation change: Support for new fingerprints.
+		var/list/found_fingerprints = GET_ATOM_FINGERPRINTS(scanned_atom)
+		if(length(found_fingerprints))
+			det_data[DETSCAN_CATEGORY_FINGERS] = list()
+			for(var/k in found_fingerprints)
+				det_data[DETSCAN_CATEGORY_FINGERS] += found_fingerprints[k]
```

Comment:
As far as I can tell you are clearing the list, then adding things to it one-by-one and then shuffling it? Is there a reason you can't just shuffle it?

Reply:
``det_data[DETSCAN_CATEGORY_FINGERS]`` doesn't exist as a list before I call ``det_data[DETSCAN_CATEGORY_FINGERS]`` = list()``

GET_ATOM_FINGERPRINTS returns the actual list (not a copy of the list) of the fingerprints. It should not be manipulated via shuffle or else issues will occur.

GET_ATOM_FINGERPRINTS is a key/value pair, so I only need to copy the value and not the key.